### ISSUE TITLE
Port twisted.test.iosim to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8064]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8064|
|Type|enhancement|
|Created|2015-10-25 03:45:35Z|
|Branch|https://github.com/twisted/twisted/tree/iosim-py3-8064|

This is a requirement for Treq to be ported.

<details><summary>Searchable metadata</summary>

```
trac-id__8064 8064
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_iosim_py3_8064 branches/iosim-py3-8064
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1445744735721711 1445744735721711
changetime__1445906499853988 1445906499853988
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45967]) Branching to iosim-py3-8064.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

This was mostly ported, but some things weren't bytes/unicode clean.

Fixed a few things, added a test to test it. Builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @hawkowl|
|-|-|

The ticket description should have been more specific, though: it was already "ported" in that it was distributed and its tests were running; this was to fix a specific issue with `writeSequence`, implements, and clean up some random twistedchecker errors.

Nevertheless, LGTM, it's all minor improvements and not too big to review; please land.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45998]) Merge iosim-py3-8064: Port the rest of twisted.test.iosim to Python 3

Author: hawkowl
Reviewer: glyph
Fixes: [#8064](https://github.com/twisted/twisted/issues/8064)
